src/basic/fd-util.c

```diff
@@ -891,21 +891,18 @@ int read_nr_open(void) {
 }
 
 int fd_get_diskseq(int fd, uint64_t *ret) {
-        uint64_t diskseq;
+        uint64_t diskseq = 0;
 
         assert(fd >= 0);
         assert(ret);
 
-        if (ioctl(fd, BLKGETDISKSEQ, &diskseq) < 0) {
+        if (ioctl(fd, BLKGETDISKSEQ, &diskseq) < 0)
                 /* Note that the kernel is weird: non-existing ioctls currently return EINVAL
                  * rather than ENOTTY on loopback block devices. They should fix that in the kernel,
                  * but in the meantime we accept both here. */
                 if (!ERRNO_IS_NOT_SUPPORTED(errno) && errno != EINVAL)
                         return -errno;
 
-                return -EOPNOTSUPP;
```

Comment:
hmm, what? why would we return a zero diskseq if the device doesn#t support it?

Reply:
To avoid all callers check for EOPNOTSUPP  and set diskseq=0 themself in this case. IIUC the rest of the code interprets diskseq=0 as "diskseq not supported by this device".

Reply:
such a change might be acceptable, but should be explicitly documented in a comment, which clarifies that diskseq=0 does not exist